src/psygnal/_signal.py

```diff
@@ -1105,10 +1106,21 @@ def resume(self, reducer: ReducerFunc | None = None, initial: Any = _NULL) -> No
         if not getattr(self, "_args_queue", None):
             return
         if reducer is not None:
-            if initial is _NULL:
-                args = reduce(reducer, self._args_queue)
+            if (
+                hasattr(reducer, "__annotations__")
+                and len(reducer.__annotations__) == 2
+            ):
+                args = cast(Callable[[List[tuple]], tuple], reducer)(self._args_queue)
+            elif initial is _NULL:
+                args = reduce(
+                    cast(Callable[[tuple, tuple], tuple], reducer), self._args_queue
+                )
             else:
-                args = reduce(reducer, self._args_queue, initial)
+                args = reduce(
+                    cast(Callable[[tuple, tuple], tuple], reducer),
+                    self._args_queue,
+                    initial,
+                )
```

Comment:
```suggestion
            if len(inspect.signature(reducer).parameters) == 1:
                reducer = cast(Callable[[Iterable[tuple]], tuple], reducer)
                args = reducer(self._args_queue)
            else:
                reducer = cast(Callable[[tuple, tuple], tuple], reducer)
                if initial is _NULL:
                    args = reduce(reducer, self._args_queue)
                else:
                    args = reduce(reducer, self._args_queue, initial)
```

are you ok with this instead?

Reply:
Yes. I wrote more complex cod because of performance worries

Reply:
I needto apply it manually, so I will do it later

Reply:
ok, thanks!